ThirdParty/SQLite.swift/Source/Core/Statement.swift

```diff
@@ -271,8 +271,13 @@ extension Cursor : SequenceType {
 
     public func generate() -> AnyGenerator<Binding?> {
         var idx = 0
-        return anyGenerator {
-            idx >= self.columnCount ? Optional<Binding?>.None : self[idx++]
+        return AnyGenerator {
+            if idx >= self.columnCount {
+                return Optional<Binding?>.None
+            } else {
+                idx += 1
+                return self[idx - 1] // This looks a bit odd but only way to compile it seems
```

Comment:
I assume the compile failure happens when trying to assign `self[idx]` to a variable before incrementing? A cleaner way to do this would be using `defer` as mentioned above, which doesn't require a temp variable:

```
defer { idx += 1 }
return self[idx]
```